### PR TITLE
Update MLClient to the latest interface and making use of promoted properties

### PR DESCRIPTION
**Pull request by luigiw**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb [hunk 1]
```diff
@@ -18,9 +19,9 @@
    "outputs": [],
    "source": [
     "#enter details of your AML workspace\n",
-    "subscription_id = '<SUBSCRIPTION_ID>'\n",
-    "resource_group = '<RESOURCE_GROUP>'\n",
-    "workspace = '<AML_WORKSPACE_NAME>'"
+    "subscription_id = 'd128f140-94e6-4175-87a7-954b9d27db16'\n",
+    "resource_group = 'ragargeastus2euap'\n",
+    "workspace = 'hanchi-test'"
    ]
```
Comment (balapv) [inline sdk/endpoints/online/sample/online-inferencing-sample.ipynb:25]: Can you please remove workspace and RG details from the notebook.

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb [hunk 2]
```diff
@@ -30,7 +31,7 @@
    "outputs": [],
    "source": [
     "#get a handle to the workspace\n",
-    "ml_client = MLClient(subscription_id, resource_group, workspace)"
+    "ml_client = MLClient(InteractiveBrowserCredential(), subscription_id, resource_group, workspace)"
```
Comment (balapv) [inline sdk/endpoints/online/sample/online-inferencing-sample.ipynb:34]: why explicitly specify InteractiveBrowserCredential? What happens if I don't specify?

File: sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb [hunk 3]
```diff
@@ -40,10 +41,10 @@
    "outputs": [],
    "source": [
     "job = CommandJob(\n",
-    "    code=Code(local_path=\"./src\"), #local path where the code is stored\n",
+    "    code_local_path=\"./src\", #local path where the code is stored\n",
     "    command = \"pip install -r requirements.txt && python main.py --iris-csv ${{inputs.iris_csv}} --epochs ${{inputs.epochs}} --lr ${{inputs.lr}}\",\n",
     "    inputs = {\n",
-    "        'iris_csv': 'https://azuremlexamples.blob.core.windows.net/datasets/iris.csv',\n",
+    "        'iris_csv': JobInput(file='https://azuremlexamples.blob.core.windows.net/datasets/iris.csv'),\n",
```
Comment (balapv) [inline sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb:47]: are we going to use JobInputs for all non-primitive types?

**Review by balapv — COMMENTED** [review]
Added my comments

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb (hunk 2 above)
Comment (luigiw) [reply sdk/endpoints/online/sample/online-inferencing-sample.ipynb:34]: `crendential` is now a required argument of MLCLient constructor. This aligns with other Azure SDKs and also what AK has been pushing for.

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb (hunk 1 above)
Comment (luigiw) [reply sdk/endpoints/online/sample/online-inferencing-sample.ipynb]: Updated now. Missed one local commit.

File: sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb (hunk 3 above)
Comment (luigiw) [reply sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb:47]: Yes, this simplifies the job inputs/outputs experience for customers. Let me know if you see any issue with this.

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb (hunk 2 above)
Comment (luigiw) [reply sdk/endpoints/online/sample/online-inferencing-sample.ipynb:34]: `crendential` is now a required argument of MLCLient constructor. This aligns with other Azure SDKs and also what AK has been pushing for.

Comment (balapv) [reply sdk/endpoints/online/sample/online-inferencing-sample.ipynb:34]: OK. Sounds good. What other options do I have? What if i run it in a scheduled job? I cannot use interactive there.

File: sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb (hunk 3 above)
Comment (balapv) [reply sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb:47]: Couple of questions:
1. How will user differentiate on when to use job inputs vs just use the primitives. For e.g. will an array of strings be a primitive or should it be decorated by a Job Input type?
2. Apart from file, what else can be in a job input constructor?
3. Is Job input the only way for non-primitives? Or can I just do it the old way as well?

File: sdk/endpoints/online/sample/online-inferencing-sample.ipynb (hunk 2 above)
Comment (luigiw) [reply sdk/endpoints/online/sample/online-inferencing-sample.ipynb:34]: Yeah, you can use any credential that's supported by `azure-identity` package.  E.g. this is how you authenticate using a service principle. https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate-service-principals#authenticate-using-environment-variables

File: sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb (hunk 3 above)
Comment (luigiw) [reply sdk/jobs/single-step/pytorch/iris/pytorch-iris.ipynb:47]: 1. JobInput is for artifact inputs, e.g. dataset and uri. Primitive inputs are passed in as they are. Lists and objects are not supported yet. Once they are, I foresee them being passed in the same way as primitives.
2. See the JobInput spec here. https://github.com/Azure/azureml_run_specification/blob/master/specs/python-sdk/job-python.md
3. Yes, JobInput is the only way.

**Pull request merged**